File: contracts/requests/DeviceRequests.ts
export interface DiscoverDevicesRequest {
  timeoutMs?: number;
  includeOffline?: boolean;
}

export interface PairDeviceRequest {
  deviceId: string;
  pairingCode?: string | null;
}

export interface UnpairDeviceRequest {
  deviceId: string;
}

export interface ConnectDeviceRequest {
  deviceId: string;
}

File: contracts/requests/DeviceRequests.kt
package contracts.requests

import kotlinx.serialization.Serializable

@Serializable
data class DiscoverDevicesRequest(
    val timeoutMs: Long = 5000,
    val includeOffline: Boolean = false
)

@Serializable
data class PairDeviceRequest(
    val deviceId: String,
    val pairingCode: String? = null
)

@Serializable
data class UnpairDeviceRequest(
    val deviceId: String
)

@Serializable
data class ConnectDeviceRequest(
    val deviceId: String
)
